docs/auto-create-collections-by-product-type-or-vendor/script.liquid

```diff
@@ -0,0 +1,347 @@
+        {% capture mutation %}
+          mutation {
+            collectionCreate(
+              input: {
+                title: {{ product_type | json }}
+                ruleSet: {
+                  appliedDisjunctively: false
+                  rules: [
+                    {
+                      column: TYPE
+                      relation: EQUALS
+                      condition: {{ product_type | json }}
+                    }
+                  ]
+                }
+              }
+            ) {
+              collection {
+                id
+                title
+                handle
+              }
+              userErrors {
+                field
+                message
+              }
+            }
+          }
+        {% endcapture %}
+
+        {
+          "action": {
+            "type": "shopify",
+            "options": {{ mutation | json }},
+            "meta": {
+              "publication_ids": {{ publication_ids | json }}
+            }
+          }
+        }
+      {% endif %}
+    {% endfor %}
+  {% endif %}
+
+  {% if create_collections_by_vendor %}
+    {% for vendor in vendors %}
+      {% capture query %}
+        query {
+          collections(
+            first: 1
+            query: {{ vendor | json | prepend: "title:" | json }}
+          ) {
+            edges {
+              node {
+                id
+                title
+                handle
+              }
+            }
+          }
+        }
+      {% endcapture %}
+
+      {% assign result = query | shopify %}
+
+      {% if result.data.collections.edges != blank %}
+        {% log
+          message: "A collection title matching this vendor already exists; skipping.",
+          vendor: vendor,
+          collection: result.data.collections.edges.first.node
+        %}
+
+      {% else %}
+        {% capture mutation %}
+          mutation {
+            collectionCreate(
+              input: {
+                title: {{ vendor | json }}
+                ruleSet: {
+                  appliedDisjunctively: false
+                  rules: [
+                    {
+                      column: VENDOR
+                      relation: EQUALS
+                      condition: {{ vendor | json }}
+                    }
+                  ]
+                }
+              }
+            ) {
+              collection {
+                id
+                title
+                handle
+              }
+              userErrors {
+                field
+                message
+              }
+            }
+          }
+        {% endcapture %}
+
+        {
+          "action": {
+            "type": "shopify",
+            "options": {{ mutation | json }},
+            "meta": {
+              "publication_ids": {{ publication_ids | json }}
+            }
+          }
+        }
+      {% endif %}
+    {% endfor %}
+  {% endif %}
+
+{% elsif event.topic == "mechanic/actions/perform" %}
+  {% assign event_data = event.data %}
```

Comment:
You've got `action` to use here, as an environment variable offered by this event topic. In an effort to encourage its use, please use that variable here (and `action.run.ok` etc), instead of digging into `event.data`.

Reply:
No problem, I went this route because the event preview assignment seemed weird if I had used `{% assign action = action_result | parse_json %}`. Not sure exactly why that felt weird.

And upon reflection, the library could use some more examples of perform action structure, so I'm happy to reformulate anything else you see that could be more standardized.